clarity around aws secrets manager implementation (#737)
astronomer/docs: astro/secrets-backend.md

apache-airflow-providers-amazon
```

- **Option 1**: Add the following environment variables to your project's `.env` file:

```dockerfile
AWS_ACCESS_KEY_ID="<your-aws-access-key-id>"
AWS_SECRET_ACCESS_KEY="<your-aws-secret-access-key>"
AWS_DEFAULT_REGION="<your-aws-region>"
AIRFLOW__SECRETS__BACKEND="airflow.providers.amazon.aws.secrets.secrets_manager.SecretsManagerBackend"
AIRFLOW__SECRETS__BACKEND_KWARGS='{"connections_prefix": "airflow/connections", "variables_prefix": "airflow/variables"}'
```

> - **Option 2**: Add the following environment variables to your project's `Dockerfile` (note the addition of `ENV` before the environment variables when compared to Option 1):

Reviewer: @aspain Think we actually need to give these two options? If you had to pick, which do you see as better practice? Assume `.env`. I feel like we pick the 1 that we'd generally recommend and then link out to our Env Vars doc for more info

Author: @paolaperaza agreed, let's go with `.env` given the new CLI functionality. If we link to https://docs.astronomer.io/astro/environment-variables I think that page could also use some love around the `.env` option

Author: @paolaperaza here's the specific commit for this piece: https://github.com/astronomer/docs/pull/737/commits/3df5b634710ca3d7df22f71890cd225f89981490

#### Step 4: Deploy to Astro

> Once you've confirmed that the integration with AWS Secrets Manager works locally, you can now add these environment variables to your Deployment on Astro.

Reviewer 2: ```suggestion
Once you've confirmed that the integration with AWS Secrets Manager works locally, you can now add these environment variables to your Astro Deployment.
```

- **Option 1**: In the Cloud UI, manually add the same environment variables found in your `.env` or `Dockerfile` file to your Deployment [environment variables](https://docs.astronomer.io/astro/environment-variables). Specify both `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` as **Secret** to ensure that your credentials are stored securely.
    
- **Option 2**: Using the [astrocloud deployment variable update](https://docs.astronomer.io/astro/cli-reference/astrocloud-deployment-variable-update) CLI command, deploy the contents of your `.env` file directly to your Cloud UI. For example:

    ```text

>     astrocloud deployment variable update --deployment-id <your-deployment-id> --load --env .env

Reviewer: Same here - I'd suggest one that we'd generally recommend and then a note or tip at the bottom that says, "you can also do this this way"

Author: 👍 let's go with the CLI command for consistency

Author: @paolaperaza here's the specific commit for this piece: https://github.com/astronomer/docs/pull/737/commits/3df5b634710ca3d7df22f71890cd225f89981490

    ```

>     - If using this command, be sure to navigate to the Cloud UI once deployed and mark your `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` as **Secret**.

Reviewer 2: ```suggestion
    - If you use this command, after deployment navigate to the Cloud UI and mark your `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` as **Secret**.
```

    :::warning

    If using Option 2 in Step 2 (using `Dockerfile` for testing these environment variables instead of `.env`):

>     - Be sure to remove the surrounding single quotation marks (`''`) from `AIRFLOW__SECRETS__BACKEND_KWARGS` and the double quotation marks (`""`) from all other environment variable values defined in your `Dockerfile`. If you add these values with the quotation marks included in your `Dockerfile`, your configuration will not work on Astro.

Reviewer 2: ```suggestion
    - Be sure to remove the surrounding single quotation marks (`''`) from `AIRFLOW__SECRETS__BACKEND_KWARGS` and the double quotation marks (`""`) from all other environment variable values defined in your `Dockerfile`. If you add these values with the quotation marks included in your `Dockerfile`, your configuration won't work on Astro.
```

>     - Delete the environment variables from your `Dockerfile` before deploying, as these environment variable values would be exposed in your repository. 

Reviewer 2: ```suggestion
    - To prevent the environment variable values from being exposed in your repository, delete the environment variables from your `Dockerfile` before deploying. 
```

- A [service account](https://cloud.google.com/iam/docs/creating-managing-service-accounts) with the [Secret Manager Secret Accessor](https://cloud.google.com/secret-manager/docs/access-control) role on Google Cloud.
- A [JSON service account key](https://cloud.google.com/iam/docs/creating-managing-service-account-keys#creating_service_account_keys) for the service account.

#### Step 1: Create an Airflow Variable or Connection in Google Cloud Secret Manager

> To start, add an Airflow variable or connection as a secret to Google Cloud Secret Manager. You can do so via the Cloud Console or the gcloud CLI. You will use this secret to test your backend's functionality in Step 3, so it can be either a real or placeholder value.

Reviewer: ```suggestion
To start, create an Airflow variable or connection in Google Cloud Secret Manager that you want to store as a secret. You can do so via the Cloud Console or the gcloud CLI.

You will use this secret to test your backend's functionality in Step 3, so it can be either a real or test value.
```
